src/main/java/com/traveltime/sdk/dto/requests/proto/Transportation.java

```diff
@@ -1,35 +1,200 @@
 package com.traveltime.sdk.dto.requests.proto;
 
+import com.igeolise.traveltime.rabbitmq.requests.RequestsCommon;
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
 
+
 public interface Transportation {
-    String getValue();
-    Integer getCode();
+    TransportationType getType();
+    RequestsCommon.Transportation getProtoMessage();
+
+    Transportation PUBLIC_TRANSPORT = PublicTransport.builder().build();
+    Transportation DRIVING_AND_PUBLIC_TRANSPORT = DrivingAndPublicTransport.builder().build();
+    Transportation WALKING_FERRY = WalkingFerry.builder().build();
+    Transportation CYCLING_FERRY = CyclingFerry.builder().build();
+    Transportation DRIVING_FERRY = DrivingFerry.builder().build();
+    Transportation WALKING = Walking.builder().build();
+    Transportation CYCLING = Cycling.builder().build();
+    Transportation DRIVING = Driving.builder().build();
```

Comment:
@bartosz-witkowski @mjanuszkiewicz-tt Are you fine with this approach or should I re-write it some other way?

Reply:
The change now is `Transportation.Modes.DRIVING_FERRY` -> `Transportation.DRIVING_FERRY`

Reply:
I could re-write it that there is no such change.

Reply:
Personally, I'd drop the builders and just use `new $Mode` and make the setters (where applicable) do `return this;`. Apart from that looks fine to me.

Reply:
> I could re-write it that there is no such change.

Oh yeah I would prefer if we didn't make any (source code) changes necessary if possible.

Reply:
Since we already use lombok here: https://projectlombok.org/features/With

Reply:
@mjanuszkiewicz-tt I could introduce `with` but everywhere else for "get me custom" we use Builder pattern.
I assume by using `with` we create  instance with default params, and if user wants to set custom param the `with` method will be used, or you meant something else?

Reply:
You assume correct.
I think that a Builder only really makes sense if there are values which can have no default, or if setting one value usually implies setting others.
In our case for transportation details I think the logical flow is "I want the default thing but to tweak a single value", so `with` makes more sense.